examples/images_contours_and_fields/image_annotated_heatmap.py

```diff
@@ -0,0 +1,313 @@
+
+# sphinx_gallery_thumbnail_number = 2
+
+##############################################################################
+#
+# A simple categorical heatmap
+# ----------------------------
+#
+# We may start by defining some data. What we need is a 2D list or array
+# which defines the data to color code. We then also need two lists or arrays
+# of categories; of course the number of elements in those lists
+# need to match the data along the respective axes.
+# The heatmap itself is an :meth:`imshow <matplotlib.axes.Axes.imshow>` plot
+# with the labels set to the categories we have.
+# Note that it is important to set both, the tick locations
+# (:meth:`set_xticks<matplotlib.axes.Axes.set_xticks>`) as well as the
+# tick labels (:meth:`set_xticklabels<matplotlib.axes.Axes.set_xticklabels>`),
+# otherwise they would become out of sync. The locations are just
+# the ascending integer numbers, while the ticklabels are the labels to show.
+# Finally we can label the data itself by creating a
+# :class:`~matplotlib.text.Text` within each cell showing the value of
+# that cell.
+#
+
+import numpy as np
+import matplotlib
+import matplotlib.pyplot as plt
+
+vegetables = ["cucumber", "tomato", "lettuce", "asparagus",
+              "potato", "wheat", "barley"]
+farmers = ["Farmer Joe", "Upland Bros.", "Smith Gardening",
+           "Agrifun", "Organiculture", "BioGoods Ltd.", "Cornylee Corp."]
+
+harvest = np.array([[0.8, 2.4, 2.5, 3.9, 0.0, 4.0, 0.0],
+                    [2.4, 0.0, 4.0, 1.0, 2.7, 0.0, 0.0],
+                    [1.1, 2.4, 0.8, 4.3, 1.9, 4.4, 0.0],
+                    [0.6, 0.0, 0.3, 0.0, 3.1, 0.0, 0.0],
+                    [0.7, 1.7, 0.6, 2.6, 2.2, 6.2, 0.0],
+                    [1.3, 1.2, 0.0, 0.0, 0.0, 3.2, 5.1],
+                    [0.1, 2.0, 0.0, 1.4, 0.0, 1.9, 6.3]])
+
+
+fig, ax = plt.subplots()
+im = ax.imshow(harvest)
+
+# We want to show all ticks...
+ax.set_xticks(np.arange(len(farmers)))
+ax.set_yticks(np.arange(len(vegetables)))
+# ... and label them with the respective list entries
+ax.set_xticklabels(farmers)
+ax.set_yticklabels(vegetables)
+
+# Rotate the tick labels and set their alignment.
+plt.setp(ax.get_xticklabels(), rotation=45, ha="right",
+         rotation_mode="anchor")
+
+# Loop over data dimensions and create text annotations.
+for i in range(len(vegetables)):
+    for j in range(len(farmers)):
+        text = ax.text(j, i, harvest[i, j],
+                       ha="center", va="center", color="w")
+
+ax.set_title("Harvest of local farmers (in tons/year)")
+fig.tight_layout()
+plt.show()
+
+
+#############################################################################
+# Using the helper function code style
+# ------------------------------------
+#
+# As discussed in the :ref:`Coding styles <coding_styles>`
+# one might want to reuse such code to create some kind of heatmap
+# for different input data and/or on different axes.
+# We might hence create a function which takes the data and the row and
```

Comment:
"We might hence" is pretty verbose.  -> "we create a..."

Usually only use "which" after a comma.  "which"->"that" 

Suggest: "We create a function that takes the data and the row and column labels as input, and allows arguments that are used to customize the plot"